### PR TITLE
Use recwarn everywhere

### PR DESCRIPTION
**Pull request by untitaker**
...instead of custom fixture. Also assert that no warnings are left over
after the test.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -126,8 +126,9 @@ def inner(name):
     return inner
 
 
-@pytest.fixture
-def catch_deprecation_warnings():
-    import warnings
-    warnings.simplefilter('default', category=DeprecationWarning)
-    return lambda: warnings.catch_warnings(record=True)
+@pytest.fixture(autouse=True)
+def catch_deprecation_warnings(recwarn, request):
```
Comment (ThiefMaster) [inline tests/conftest.py]: I think `yield_fixture` would be cleaner and more readable:

```python
@pytest.yield_fixture(autouse=True)
def catch_deprecation_warnings(recwarn):
    yield
    assert not recwarn.list
```

**Comment by untitaker** [comment]
@ThiefMaster @davidism Please review.

File: tests/conftest.py (hunk 1 above)
Comment (untitaker) [inline tests/conftest.py]: Interesting, wasn't aware of that.

**Comment by untitaker** [comment]
Interesting, we're getting a ResourceWarning somewhere.

**Pull request merged**